src/rdkafka_topic.c

```diff
@@ -657,7 +657,7 @@ static void rd_kafka_topic_assign_uas (rd_kafka_itopic_t *rkt,
 	rd_kafka_msg_t *rkm, *tmp;
 	rd_kafka_msgq_t uas = RD_KAFKA_MSGQ_INITIALIZER(uas);
 	rd_kafka_msgq_t failed = RD_KAFKA_MSGQ_INITIALIZER(failed);
-	int cnt;
+	int cnt, failed_cnt = 0;
```

Comment:
This is the slow-path so you don't really need to do this optimization here. Suggest removing this patch.

Reply:
Well, the if (failed_cnt > 0) check at the very end is always executed, so that matters somewhat I guess.  It also moves us closer to a world where the count is just protected by the queue lock instead of using atomic operations.  I'm not sure what downside you see, but I'll remove this files changes for now to get the rest in.